node/network/statement-distribution/src/lib.rs

```diff
@@ -508,7 +508,7 @@ enum LargeStatementStatus {
 struct FetchingInfo {
 	/// All peers that send us a `LargeStatement` or a `Valid` statement for the given
 	/// `CandidateHash`, together with their originally sent messages.
-	available_peers: HashMap<PeerId, Vec<protocol_v1::StatementDistributionMessage>>,
+	available_peers: IndexMap<PeerId, Vec<protocol_v1::StatementDistributionMessage>>,
```

Comment:
my first though was it is used to ensure determinism in tests, is that true?

Reply:
no, we iterate over that map in the end to finally process arrived messages. By using `IndexMap` we ensure the messages will be forwarded in the order they were received, which is desirable as we reward the first peer more than later peers.

Reply:
I see, would you mind adding this as doc comments then?